homeassistant/components/fan/__init__.py

```diff
@@ -46,17 +48,30 @@
 DIRECTION_REVERSE = "reverse"
 
 ATTR_SPEED = "speed"
+ATTR_PERCENTAGE = "percentage"
 ATTR_SPEED_LIST = "speed_list"
 ATTR_OSCILLATING = "oscillating"
 ATTR_DIRECTION = "direction"
 
+INVALID_SPEEDS_FILTER = {"on", "auto", "smart"}
```

Comment:
Does this filter cover all the existing fan platforms with invalid speeds?

Reply:
It does for at least the ones I could find and this list https://github.com/home-assistant/architecture/issues/127#issuecomment-678999782